cmd/verify/network/cmd.go

```diff
@@ -106,43 +106,42 @@ func init() {
 
 func run(cmd *cobra.Command, _ []string) {
 	r := rosa.NewRuntime().WithAWS().WithOCM()
+	var cluster *cmv1.Cluster
 	defer r.Cleanup()
 
-	if cmd.Flags().Changed(subnetIDsFlag) {
-		if cmd.Flags().Changed(clusterFlag) {
-			r.Reporter.Errorf("One and only one of %s or %s is required",
-				clusterFlag, subnetIDsFlag)
-			os.Exit(1)
-		}
+	if cmd.Flags().Changed(clusterFlag) {
+		cluster = r.FetchCluster()
 
-		region, err := aws.GetRegion(arguments.GetRegion())
-		if err != nil {
-			r.Reporter.Errorf("Error getting region: %v", err)
-			os.Exit(1)
-		}
-		args.region = region
-	} else if cmd.Flags().Changed(clusterFlag) {
-		if cmd.Flags().Changed(subnetIDsFlag) {
-			r.Reporter.Errorf("One and only one of %s or %s is required",
-				clusterFlag, subnetIDsFlag)
+		if cluster.Status() != nil && cluster.Status().State() != cmv1.ClusterStateReady {
+			r.Reporter.Errorf("Cluster state is not ready: %v", cluster.Status().State())
 			os.Exit(1)
 		}
-		cluster := r.FetchCluster()
+	}
 
-		if cluster.Status() != nil && cluster.Status().State() != cmv1.ClusterStateReady {
-			r.Reporter.Errorf("Cluster state is not ready: %v", cluster.Status().State())
+	if !cmd.Flags().Changed(subnetIDsFlag) {
+		if cluster != nil {
+			args.subnetIDs = cluster.AWS().SubnetIDs()
+			if len(args.subnetIDs) == 0 {
+				r.Reporter.Errorf("No subnets on cluster '%s'", cluster.Name())
+				os.Exit(1)
+			}
+		} else {
+			r.Reporter.Errorf("At least one subnet IDs is required")
 			os.Exit(1)
 		}
+	}
 
-		args.subnetIDs = cluster.AWS().SubnetIDs()
-		if len(args.subnetIDs) == 0 {
-			r.Reporter.Errorf("No subnets on cluster")
+	if cmd.Flags().Changed("region") {
```

Comment:
Getting the region value, can we flatten out this whole block. Bring it out to its own func, and return early. Getting rid of the `else` and `else if`.

Reply:
this doesn't seem complex enough to pull out into separate func, it's pretty compact and legible imo

Reply:
I agree this current block is not complex, but keeping indentation to a minimum is a best practice for golang. 

ROSA is full of nested if's and else statements. We need to start cutting back on them and follow the best practices. 

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow
https://go.dev/doc/effective_go

Reply:
updated, thanks!